clients/include/testing_geam_batched.hpp

```diff
@@ -0,0 +1,534 @@
+        {
+            if((lda == ldc) && (transA == rocblas_operation_none))
+                CHECK_HIP_ERROR(dC_in_place.transfer_from(hA));
+            //
+            CHECK_ROCBLAS_ERROR(rocblas_set_pointer_mode(handle, rocblas_pointer_mode_host));
+            auto status_h = rocblas_geam_batched<T>(handle,
+                                                    transA,
+                                                    transB,
+                                                    M,
+                                                    N,
+                                                    &alpha,
+                                                    dC_in_place.ptr_on_device(),
+                                                    lda,
+                                                    &beta,
+                                                    dB.ptr_on_device(),
+                                                    ldb,
+                                                    dC_in_place.ptr_on_device(),
+                                                    ldc,
+                                                    batch_count);
+
+            if(lda != ldc || transA != rocblas_operation_none)
+            {
+                EXPECT_ROCBLAS_STATUS(status_h, rocblas_status_invalid_size);
+            }
+            else
+            {
+                CHECK_HIP_ERROR(hC_1.transfer_from(dC));
+                // dA was clobbered by dC_in_place, so copy hA back to dA
+                CHECK_HIP_ERROR(dA.transfer_from(hA));
+
+                // reference calculation
+                for(int i3 = 0; i3 < batch_count; i3++)
+                {
+                    auto hA_copy_p = hA_copy[i3];
+                    auto hB_copy_p = hB_copy[i3];
+                    auto hC_gold_p = hC_gold[i3];
+
+                    for(int i1 = 0; i1 < M; i1++)
+                    {
+                        for(int i2 = 0; i2 < N; i2++)
+                        {
+                            hC_gold_p[i1 + i2 * ldc]
+                                = alpha * hA_copy_p[i1 * inc1_A + i2 * inc2_A]
+                                  + beta * hB_copy_p[i1 * inc1_B + i2 * inc2_B];
+                        }
+                    }
+                }
+
+                if(arg.unit_check)
+                {
+                    unit_check_general<T>(M, N, ldc, hC_gold, hC_1, batch_count);
+                }
+
+                if(arg.norm_check)
+                {
+                    rocblas_error
+                        = norm_check_general<T>('F', M, N, ldc, hC_gold, hC_1, batch_count);
+                }
+            }
+        }
+
+        // inplace check for dC == dB
+        {
+            if((ldb == ldc) && (transB == rocblas_operation_none))
+                CHECK_HIP_ERROR(dC_in_place.transfer_from(hB));
+
+            CHECK_ROCBLAS_ERROR(rocblas_set_pointer_mode(handle, rocblas_pointer_mode_host));
+            auto status_h = rocblas_geam_batched<T>(handle,
```

Comment:
Don't worry about it now but calling directly conflicts with Lee's proposed EXPECT_ROCBLAS_STATUS being a lambda function call with signal and exception handling.    This will still work just without the handling.